src/test/java/io/cdap/plugin/successfactors/source/transport/SuccessFactorsSourceTest.java

```diff
@@ -0,0 +1,503 @@
+        minTimes = 0;
+      }
+    };
+    try {
+      successFactorsSource = new SuccessFactorsSource(pluginConfig);
+      Map<String, Object> plugins = new HashMap<>();
+      MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+      successFactorsSource.configurePipeline(mockPipelineConfigurer);
+      Assert.fail("Connection exception is not thrown if valid connection is provided");
+    } catch (ValidationException te) {
+      List<ValidationFailure> failures = te.getFailures();
+      Assert.assertEquals(1, failures.size());
+    }
+  }
+
+  private List<SuccessFactorsInputSplit> getSplits() {
+    List<SuccessFactorsInputSplit> splits = new ArrayList<>();
+
+    splits.add(new SuccessFactorsInputSplit(101, 120, 20));
+    splits.add(new SuccessFactorsInputSplit(121, 140, 20));
+    splits.add(new SuccessFactorsInputSplit(141, 160, 20));
+    splits.add(new SuccessFactorsInputSplit(161, 180, 20));
+    splits.add(new SuccessFactorsInputSplit(181, 200, 20));
+
+    return splits;
+  }
+
+  @Test
+  public void testPrepareRun() throws Exception {
+    successFactorsService = new SuccessFactorsService(pluginConfig, null);
+    successFactorsPartitionBuilder = new SuccessFactorsPartitionBuilder();
+    successFactorsTransporter = new SuccessFactorsTransporter("username", "password");
+    pluginConfigBuilder = SuccessFactorsPluginConfig.builder()
+      .referenceName("unit-test-ref-name")
+      .baseURL("http://localhost")
+      .entityName("entity name")
+      .username("username")
+      .password("password")
+      .selectOption("col1,col2,   \n  parent/col1,\r       col3     ")
+      .filterOption("$topeq2")
+      .splitCount(5)
+      .numRowsToFetch(100L)
+      .batchSize(100L)
+      .skipRowCount(0L);
+
+    pluginConfig = pluginConfigBuilder.build();
+
+    new Expectations(SuccessFactorsService.class, SuccessFactorsTransporter.class) {
+      {
+        context.getOutputSchema();
+        result = getPluginSchema();
+        minTimes = 0;
+
+        successFactorsTransporter.callSuccessFactors(null, anyString, anyString);
+        result = getResponseContainer();
+        minTimes = 0;
+
+        successFactorsPartitionBuilder.buildSplit(50, 1000L,
+                                                  pluginConfig.getSkipRowCount(), pluginConfig.getSplitCount(),
+                                                  pluginConfig.getBatchSize());
+
+        result = getSplits();
+        minTimes = 0;
+
+        successFactorsService.getEncodedServiceMetadata();
+        result = "encodeMetadataString";
+        minTimes = 0;
+      }
+
+    };
+    successFactorsSource = new SuccessFactorsSource(pluginConfig);
+    successFactorsSource.prepareRun(context);
+    Map<String, Object> plugins = new HashMap<>();
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+    schema = mockPipelineConfigurer.getOutputSchema();
+    Assert.assertNull(schema);
```

Comment:
What is the purpose of this test case if we are validating null in the end?

Reply:
Added a valid assertion.